libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -299,6 +299,21 @@ uint8_t AP_GPS::num_sensors(void) const
     }
 }
 
+bool AP_GPS::is_healthy(bool record_log, bool send_feedback) {
+    return is_healthy(0, record_log, send_feedback); //could be overloaded with defaults
+}
+
+bool AP_GPS::is_healthy(uint8_t instance, bool record_log, bool send_feedback) {
+    uint32_t tnow = AP_HAL::millis();
+    if (drivers[instance] != nullptr && state[instance].status != NO_GPS &&
+      tnow - timing[instance].last_message_time_ms < get_rate_ms(instance) &&
```

Comment:
```get_rate_ms()``` can't be used like that. It's a user parameter to request the desired update rate, but it is only forced on u-blox, for any other driver you don't know what its at. There is also timing jitter (20ms or more) to consider which would cause you to sometimes not healthy() until the next message comes in. You need to add a buffer for timing jitter.

(On a different note I need to add a bunch of disclaimers in comments to ```get_rate_ms()``` because it's *only* a parameter accessors, and provides no guarantees about what a given backend is doing. People keep trying to use it like it's a truth/accurate value and that is not a guarantee it makes.